### ISSUE TITLE
vim 8.2.1552 segfaults in setManualFoldForWin

### ISSUE THREAD

**Issue by mgedmin** [vim/vim#6830]
**Describe the bug**

vim segfaulted on me while I was editing a YAML file.  Twice.  It happened when I pressed Backspace in insert mode.  Here's a backtrace from the core dump:

```
Reading symbols from /home/mg/src/vim/src/vim...

warning: Unexpected size of section `.reg-xstate/1028183' in core file.
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
Core was generated by `/home/mg/src/vim/src/vim'.
Program terminated with signal SIGSEGV, Segmentation fault.

warning: Unexpected size of section `.reg-xstate/1028183' in core file.
#0  0x00007f0daf40555b in kill () at ../sysdeps/unix/syscall-template.S:78
78	../sysdeps/unix/syscall-template.S: Toks failas ar aplankas neegzistuoja.
(gdb) bt
#0  0x00007f0daf40555b in kill () at ../sysdeps/unix/syscall-template.S:78
#1  0x000055fa6f9c0ddf in may_core_dump () at os_unix.c:3426
#2  0x000055fa6f9c0d7f in mch_exit (r=1) at os_unix.c:3392
#3  0x000055fa6fb3d1c1 in getout (exitval=1) at main.c:1686
#4  0x000055fa6f981db4 in preserve_exit () at misc1.c:2189
#5  0x000055fa6f9be50c in deathtrap (sigarg=11) at os_unix.c:1111
#6  0x00007f0daf9143c0 in <signal handler called> () at /lib/x86_64-linux-gnu/libpthread.so.0
#7  0x000055fa6f926000 in setManualFoldWin (wp=0x55fa70d58ce0, lnum=3, opening=1, recurse=0, donep=0x7ffd361d4b54) at fold.c:1318
#8  0x000055fa6f925f2f in setManualFold (lnum=66, opening=1, recurse=0, donep=0x7ffd361d4b54) at fold.c:1273
#9  0x000055fa6f924713 in foldOpenCursor () at fold.c:456
#10 0x000055fa6f8beda2 in ins_bs (c=-25195, mode=1, inserted_space_p=0x7ffd361d4c20) at edit.c:4356
#11 0x000055fa6f8b96da in edit (cmdchar=111, startln=1, count=0) at edit.c:942
#12 0x000055fa6f99fbf5 in invoke_edit (cap=0x7ffd361d4dd0, repl=0, cmd=111, startln=1) at normal.c:7105
#13 0x000055fa6f99e832 in n_opencmd (cap=0x7ffd361d4dd0) at normal.c:6396
#14 0x000055fa6f9a071f in nv_open (cap=0x7ffd361d4dd0) at normal.c:7482
#15 0x000055fa6f994854 in normal_cmd (oap=0x7ffd361d4e70, toplevel=1) at normal.c:1091
#16 0x000055fa6fb3cdbc in main_loop (cmdwin=0, noexmode=0) at main.c:1478
#17 0x000055fa6fb3c0b4 in vim_main2 () at main.c:868
#18 0x000055fa6fb3b7a7 in main (argc=1, argv=0x7ffd361d5088) at main.c:412
```

and here are some local variables in frame 7:

```
(gdb) frame 7
#7  0x000055fa6f926000 in setManualFoldWin (wp=0x55fa70d58ce0, lnum=3, opening=1, recurse=0, donep=0x7ffd361d4b54) at fold.c:1318
1318			next = fp->fd_top + off;
(gdb) p fp
$1 = (fold_T *) 0x0
```

**To Reproduce**

I haven't yet been able to reproduce this with `vim --clean`.  With [my vim configuration](https://github.com/mgedmin/dotvim) I can reproduce if I edit a YAML file to look like this:

```yaml
some_list:
  - one
  -
    |-
  - two
```

(`|` indicating the cursor position), and then press Backspace.

**Expected behavior**

No segfault, thank you very much!

**Environment (please complete the following information):**
 - Vim version: 8.2.1558
 - OS: Ubuntu 20.04
 - Terminal: GNOME Terminal

**Additional context**

git bisect blames commit 64f37d309025a65210dbc33823ec9ec5d547775f

**Comment by chrisbra** [comment]
so #6831 fixes it, right?

**Comment by mgedmin** [comment]
[Yes](https://github.com/vim/vim/pull/6831#issuecomment-684474446).

**Comment by chrisbra** [comment]
okay, closing than for #6831

**Comment by mgedmin** [comment]
Fixed by 8.2.1561 (although I actually tested 8.2.1563, but the other two patches seem unrelated).
